main.go

```diff
@@ -4,39 +4,207 @@ import (
+		},
+		&sensu.PluginConfigOption{
+			Path:      "max-bytes",
+			Env:       "CHECK_LOG_MAX_BYTES",
+			Argument:  "max-bytes",
+			Shorthand: "b",
+			Default:   int64(0),
+			Usage:     "Max number of bytes to read (0 means unlimited).",
+			Value:     &plugin.MaxBytes,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "events-api-url",
+			Env:       "CHECK_LOG_EVENTS_API_URL",
+			Argument:  "events-api-url",
+			Shorthand: "u",
+			Default:   "http://localhost:3031/events",
+			Usage:     "Agent Events API URL.",
+			Value:     &plugin.EventsAPI,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "ignore-initial-run",
+			Env:       "CHECK_LOG_IGNORE_INITIAL_RUN",
+			Argument:  "ignore-initial-run",
+			Shorthand: "I",
+			Default:   false,
+			Usage:     "Suppresses alerts for any matches found on the first run of the plugin.",
+			Value:     &plugin.IgnoreInitialRun,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "check-name-tamplate",
+			Env:       "CHECK_LOG_CHECK_NAME_TEMPLATE",
+			Argument:  "check-name-template",
+			Shorthand: "t",
+			Default:   defaultNameTemplate,
+			Usage:     "Check name to use in generated events",
+			Value:     &plugin.CheckNameTemplate,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "disable-event-generation",
+			Env:       "CHECK_LOG_CHECK_DISABLE_EVENT_GENERATION",
+			Argument:  "disable-event-generation",
+			Shorthand: "D",
+			Default:   false,
+			Usage:     "Disable event generation, send results to stdout instead.",
+			Value:     &plugin.DisableEvent,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "reset-state",
+			Env:       "CHECK_LOG_RESET_STATE",
+			Argument:  "reset-state",
+			Shorthand: "r",
+			Default:   false,
+			Usage:     "Allow automatic state reset if match expression changes, instead of failing.",
+			Value:     &plugin.EnableStateReset,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "inverse-match",
+			Env:       "CHECK_LOG_INVERSE_MATCH",
+			Argument:  "inverse-match",
+			Shorthand: "i",
+			Default:   false,
+			Usage:     "Inverse match, only generate alert event if no lines match.",
+			Value:     &plugin.InverseMatch,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "verbose",
+			Argument:  "verbose",
+			Shorthand: "v",
+			Default:   false,
+			Usage:     "Verbose output, useful for testing.",
+			Value:     &plugin.Verbose,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "dry-run",
+			Argument:  "dry-run",
+			Shorthand: "n",
+			Default:   false,
+			Usage:     "Suppress generation of events and report intended actions instead. (implies verbose)",
+			Value:     &plugin.DryRun,
+		},
+	}
 )
 
 // State represents the state file offset
 type State struct {
-	Offset json.Number `json:"offset"`
+	Offset       json.Number `json:"offset"`
+	LastTime     int64       `json:"last_time"`
+	ModTime      int64       `json:"mod_time"`
+	MatchExpr    string      `json:"match_expr"`
+	InverseMatch bool        `json:"inverse_match"`
```

Comment:
It doesn't make sense to me that these fields are represented here, since they are supplied by the tool. Can you explain?

Reply:
Actually the regexp golang implementation doesn't allow for inverse match syntax. The regexp package implements a subset of the full re syntax.. and inversion isn't part of the supported subset. Ugh.. i was that was not the case, but from my testing, I couldn't make inverse matching work.. regexp threw errors... even in golang playground testing.

Reply:
Oh in the state file?  Oh that's because if you change the matching condition you want you've basically invalidated the stored state.
Let's say you create a check and want to match for `Error` and your regexp is case sensitive when you really wanted to match for `error` case-insensitive.
You go back and edit the MatchExpr accordingly.  
You just invalided the state file because you need to reprocess the log file again from the beginning for the new match.